### PR TITLE
rpc truncate suggestion

### PR DESCRIPTION
**Pull request by zhangsoledad**

### CODE REVIEW

File: chain/src/chain.rs [hunk 1]
```diff
@@ -194,71 +190,67 @@ impl ChainService {
...
-            let columns = vec![
-                COLUMN_INDEX,
-                COLUMN_BLOCK_HEADER,
-                COLUMN_BLOCK_BODY,
-                COLUMN_BLOCK_EXT,
-                COLUMN_BLOCK_PROPOSAL_IDS,
-                COLUMN_BLOCK_EPOCH,
-            ];
-            for column in columns {
-                db_txn.delete(column, cursor.as_slice())?;
```
Comment (zhangsoledad) [inline chain/src/chain.rs:225]: db_txn.delete(column, cursor.as_slice())?; for `COLUMN_BLOCK_BODY` is incorrect.

see https://github.com/nervosnetwork/ckb/pull/2120/files#diff-ed3597f2dbe252bad565f2898a1c8bd7R126-R141

File: tx-pool/src/process.rs [hunk 2]
```diff
@@ -508,9 +507,7 @@ impl TxPoolService {
         let config = tx_pool.config;
         let snapshot = Arc::clone(&tx_pool.snapshot);
         let last_txs_updated_at = Arc::new(AtomicU64::new(0));
-        let mut new_pool = TxPool::new(config, snapshot, last_txs_updated_at);
-        let old_pool = tx_pool.deref_mut();
-        ::std::mem::swap(old_pool, &mut new_pool);
+        *tx_pool = TxPool::new(config, snapshot, last_txs_updated_at);
```
Comment (zhangsoledad) [inline tx-pool/src/process.rs:510]: no need to swap

**Comment by keroro520** [comment]
There is a difference between `reorg` and `truncate`:
  * `reorg` moves the detached transactions into the pending pool
  * `truncate` just removes the detached transactions from the database and clear the tx-pool

I am going to cherry-pick this PR into https://github.com/nervosnetwork/ckb/pull/2064 but remove the lines https://github.com/nervosnetwork/ckb/pull/2120/files#diff-dc8555c8638c6e8bf7723ae776ce254bR245-R252

**Comment by keroro520** [comment]
> There is a difference between `reorg` and `truncate`:
> 
> * `reorg` moves the detached transactions into the pending pool
> * `truncate` just removes the detached transactions from the database and clear the tx-pool
> 
> I am going to cherry-pick this PR into #2064 but remove the lines https://github.com/nervosnetwork/ckb/pull/2120/files#diff-dc8555c8638c6e8bf7723ae776ce254bR245-R252

Commit: efc9fc8a7fa61e2dde23a24de3edb8b36cfec3d9

**Pull request closed without merging**